src/routing/Router.svelte

```diff
@@ -0,0 +1,95 @@
+<script context="module">
+  // We could grab the value of currentQuery here rather than requiring that it
+  // be passed as an argument, but it's unclear how to get an up-to-date store
+  // value from within a module.
+  export function getPath(path, query, preserveQuery = true) {
+    let returnedPath = path;
+    if (query && preserveQuery) {
+      returnedPath += `?${query}`;
+    }
+    return returnedPath;
+  }
+
+  export function navigate(...args) {
+    page.show(getPath(...args));
+  }
+</script>
+
+<script>
+  import page from 'page';
+  import { onMount } from 'svelte';
+  import { get } from 'svelte/store';
+
+  import { store, currentQuery } from '../state/store';
+
+  // Wrappers
+  import Layout from './wrappers/Layout.svelte';
+
+  // Pages
+  import Home from './pages/Home.svelte';
+  import ProbeExplore from './pages/probe/Explore.svelte';
+  import ProbeTable from './pages/probe/Table.svelte';
+  import NotFound from './pages/NotFound.svelte';
+
+
+  let component;
+  let visible = false;
+
+  function updateQueryString(query) {
+    if (window.history.pushState) {
+      const newurl = `${window.location.protocol}//${window.location.host}${window.location.pathname}?${query}`;
+      window.history.pushState({ path: newurl }, '', newurl);
+    }
+  }
+
+  // We need to reference $currentQuery in this block if we want the block to
+  // re-run whenever $currentQuery changes. If we were to call
+  // updateQueryString() with no arguments and use $currentQuery directly in the
+  // implementation of updateQueryString(), the query string would not update
+  // when the user uses the menu buttons (Channel, OS, etc.), for example.
+  $: if (visible) {
+    updateQueryString($currentQuery);
+  }
+
+  onMount(() => {
+    visible = true;
+  });
+
+  function useComponent(componentToUse, view) {
+    return function({ params: { product, item, name }}) {
+      component = componentToUse;
+
+      store.setField('route', {
+        product,
+        item,
+        name,
+        view,
+      });
+
+      store.setField('showingProbe', item === 'probe' ? true : false);
+    };
+  }
+
+  page('/', useComponent(Home));
+  page('/:product/:item/:name/explore', useComponent(ProbeExplore, 'explore'));
```

Comment:
minor nit – would you mind if we change `:item` to `:section` here + in the store? Item feels vague, and I think we're likely to add other sections for products in the future

Reply:
Also, I suspect we will have to refactor this if we have other `section`s that have a different routing structure. The URL structure will end up being the same to the end-user. This goes against my previous recommendation :) let's go with this for now, but be ready to change it later

Reply:
👍